src/ORM/DataObjectSchema.php

```diff
@@ -404,6 +404,20 @@ public function databaseIndexes($class, $aggregated = true)
         return array_merge($indexes, $this->databaseIndexes(get_parent_class($class ?? '')));
     }
 
+    /**
+     * Get an array of field names which are included in a given index.
+     * If the index doesn't exist, the array is empty.
+     */
+    public function getIndexFields(string $class, string $indexName): array
```

Comment:
This method is for reducing the amount of schema-wrangling we need to do in `DataObject::buildValidationResultForDuplicateEntry()` and seems like it could be useful for general use - but I can inline it into `DataObject::buildValidationResultForDuplicateEntry()` if reviewers don't like it.

Reply:
Just inline it, there's currently no shortage of random single use public methods which probably shouldn't exist and just add to the API surface to maintain

Reply:
Done